HMCL/build.gradle.kts

```diff
@@ -0,0 +1,273 @@
+import java.nio.file.FileSystems
+import java.nio.file.Files
+import java.security.KeyFactory
+import java.security.MessageDigest
+import java.security.Signature
+import java.security.spec.PKCS8EncodedKeySpec
+import java.util.jar.JarFile
+import java.util.jar.JarOutputStream
+import java.util.zip.ZipFile
+import java.util.zip.GZIPOutputStream
+import java.io.ByteArrayOutputStream
+import java.io.ByteArrayInputStream
+import java.net.URI
+
+import org.glavo.pack200.Pack200
+import org.tukaani.xz.LZMA2Options
+import org.tukaani.xz.XZOutputStream
+
+buildscript {
+    repositories {
+        gradlePluginPortal()
+        maven(url = "https://jitpack.io")
+    }
+    dependencies {
+        classpath("org.tukaani:xz:1.8")
+        classpath("org.glavo:pack200:0.3.0")
+    }
+}
+
+plugins {
+    application
+    id("com.github.johnrengelman.shadow") version "7.0.0"
+}
+
+val buildNumber = System.getenv("BUILD_NUMBER")?.toInt().let { number ->
+    val offset = System.getenv("BUILD_NUMBER_OFFSET")?.toInt()
+    if (number != null && offset != null) {
```

Comment:
you should assign `System.getenv("BUILD_NUMBER")` to `bulidNumber` directly when `System.getenv("BUILD_NUMBER_OFFSET")` is empty.